model/jpa/src/main/resources/META-INF/jpa-changelog-8.0.0.xml

```diff
@@ -0,0 +1,205 @@
+            <dbms type="oracle" />
+        </preConditions>
+
+        <!-- SALT was saved in tinyblob in previous version. -->
+        <!-- Can't be automatically updated for all users to new format in single UPDATE statement, so existing users will be updated on-the-fly -->
+        <update tableName="CREDENTIAL">
+            <column name="PRIORITY" value="10" />
+            <column name="SECRET_DATA" valueComputed="'{&quot;value&quot;:&quot;' || VALUE || '&quot;,&quot;salt&quot;:&quot;__SALT__&quot;}'"/>
+            <column name="CREDENTIAL_DATA" valueComputed="'{&quot;hashIterations&quot;:' || HASH_ITERATIONS || ',&quot;algorithm&quot;:&quot;' || ALGORITHM || '&quot;}'"/>
+            <where>TYPE = 'password' OR TYPE = 'password-history'</where>
+        </update>
+
+        <update tableName="CREDENTIAL">
+            <column name="PRIORITY" value="20" />
+            <column name="TYPE" value="otp" />
+            <column name="SECRET_DATA" valueComputed="'{&quot;value&quot;:&quot;' || VALUE || '&quot;}'"/>
+            <column name="CREDENTIAL_DATA" valueComputed="'{&quot;subType&quot;:&quot;totp&quot;,&quot;digits&quot;:' || DIGITS || ',&quot;period&quot;:' || PERIOD || ',&quot;algorithm&quot;:&quot;' || ALGORITHM || '&quot;}'"/>
+            <where>TYPE = 'totp'</where>
+        </update>
+
+        <update tableName="CREDENTIAL">
+            <column name="PRIORITY" value="20" />
+            <column name="TYPE" value="otp" />
+            <column name="SECRET_DATA" valueComputed="'{&quot;value&quot;:&quot;' || VALUE || '&quot;}'"/>
+            <column name="CREDENTIAL_DATA" valueComputed="'{&quot;subType&quot;:&quot;hotp&quot;,&quot;digits&quot;:' || DIGITS || ',&quot;counter&quot;:' || COUNTER || ',&quot;algorithm&quot;:&quot;' || ALGORITHM || '&quot;}'"/>
+            <where>TYPE = 'hotp'</where>
+        </update>
+
+        <!--Update format of fed_user_credential to fill secret_data and credential_data-->
+        <update tableName="FED_USER_CREDENTIAL">
+            <column name="PRIORITY" value="10" />
+            <column name="SECRET_DATA" valueComputed="'{&quot;value&quot;:&quot;' || VALUE || '&quot;,&quot;salt&quot;:&quot;__SALT__&quot;}'"/>
+            <column name="CREDENTIAL_DATA" valueComputed="'{&quot;hashIterations&quot;:' || HASH_ITERATIONS || ',&quot;algorithm&quot;:&quot;' || ALGORITHM || '&quot;}'"/>
+            <where>TYPE = 'password' OR TYPE = 'password-history'</where>
+        </update>
+
+        <update tableName="FED_USER_CREDENTIAL">
+            <column name="PRIORITY" value="20" />
+            <column name="TYPE" value="otp" />
+            <column name="SECRET_DATA" valueComputed="'{&quot;value&quot;:&quot;' || VALUE || '&quot;}'"/>
+            <column name="CREDENTIAL_DATA" valueComputed="'{&quot;subType&quot;:&quot;totp&quot;,&quot;digits&quot;:' || DIGITS || ',&quot;period&quot;:' || PERIOD || ',&quot;algorithm&quot;:&quot;' || ALGORITHM || '&quot;}'"/>
```

Comment:
Please verify that `PERIOD` does not harm MySQL / MariaDB

Reply:
Verified. Works fine on both MySQL and MariaDB. Tested with MariaDB 10.3.16 , which is the version previously having issues with the PERIOD column.